packages/shared/test/types.test.ts

```diff
@@ -0,0 +1,508 @@
+    });
+
+    it('should reject strings longer than 120 characters', () => {
+      const longString = 'a'.repeat(121);
+      expect(() => zSanitizedJustification.parse(longString)).toThrow(
+        'Justification cannot exceed 120 characters'
+      );
+    });
+
+    it('should accept strings exactly 120 characters', () => {
+      const exactString = 'a'.repeat(120);
+      expect(() => zSanitizedJustification.parse(exactString)).not.toThrow();
+    });
+  });
+
+  describe('Whitespace normalization', () => {
+    it('should trim leading and trailing whitespace', () => {
+      const result = zSanitizedJustification.parse('  Hello World  ');
+      expect(result).toBe('Hello World');
+    });
+
+    it('should normalize multiple spaces to single spaces', () => {
+      const result = zSanitizedJustification.parse('Hello    World');
+      expect(result).toBe('Hello World');
+    });
+
+    it('should normalize tabs and newlines to spaces', () => {
+      const result = zSanitizedJustification.parse('Hello\tWorld\nTest');
+      expect(result).toBe('Hello World Test');
+    });
+  });
+
+  describe('HTML/XML injection prevention', () => {
+    it('should reject HTML tags', () => {
+      const htmlPatterns = [
+        '<script>alert("xss")</script>',
+        '<div>content</div>',
+        '<iframe src="evil.com">',
+        '<img src="x" onerror="alert(1)">',
+        '<a href="javascript:alert(1)">click</a>',
+        '<style>body{display:none}</style>',
+        '<link rel="stylesheet" href="evil.css">',
+        '<meta http-equiv="refresh" content="0;url=evil.com">',
+      ];
+
+      htmlPatterns.forEach((pattern) => {
+        expect(() => zSanitizedJustification.parse(pattern)).toThrow(
+          'Justification contains invalid characters or patterns'
+        );
+      });
+    });
+
+    it('should reject XML patterns', () => {
+      const xmlPatterns = [
+        '<?xml version="1.0"?>',
+        '<!DOCTYPE html>',
+        '<![CDATA[<script>alert(1)</script>]]>',
+        '&lt;script&gt;alert(1)&lt;/script&gt;',
+        '&amp;lt;script&amp;gt;',
+      ];
+
+      xmlPatterns.forEach((pattern) => {
+        expect(() => zSanitizedJustification.parse(pattern)).toThrow(
+          'Justification contains invalid characters or patterns'
+        );
+      });
+    });
+  });
+
+  describe('JSON injection prevention', () => {
+    it('should reject JSON patterns', () => {
+      const jsonPatterns = [
+        '{"key": "value"}',
+        '{"justification": "malicious"}',
+        '["item1", "item2"]',
+        '{"type": "object", "data": {}}',
+        '"key": "value"',
+        'key: "value"',
+        '{"nested": {"object": "value"}}',
+      ];
+
+      jsonPatterns.forEach((pattern) => {
+        expect(() => zSanitizedJustification.parse(pattern)).toThrow(
+          'Justification contains invalid characters or patterns'
+        );
+      });
+    });
```

Comment:
nit: The tests that have multiple cases could use `it.each` so that each case appears as an individual test, as this makes it easier to identify which one fails (especially if you add the pattern into the test name).